spec/models/hold_spec.rb

```diff
@@ -39,4 +39,37 @@
       expect(build(:invalid_date_time_hold)).not_to be_valid
     end
   end
+
+  describe 'check_conflicting_rentals' do
+    let(:hold) { create(:hold) }
+    let(:conflicting_rental) { create(:hold_conflicting_rental) }
+    let(:future_rental) { create(:far_future_rental) }
+
+    it 'should cancel a conflicting rental' do
+      conflicting_rental
+      hold.check_conflicting_rentals
```

Comment:
does `hold.check_conflicting_rentals` replace the rental? It seems a little counter intuitive that a method named that would have a destructive action. Maybe rename it to `handle_conflicting_rentals`